app/assets/javascripts/scp-dot-plot.js

```diff
@@ -0,0 +1,296 @@
+  if (typeof window.dotPlot === 'undefined') {
+    // Consistent with e.g. IGV, Ideogram
+    ga('send', 'event', 'dot-plot', 'initialize');
+  }
+
+  // Instantiate dot plot and embed in DOM element
+  window.dotPlot = new morpheus.HeatMap(config);
+  window.dotPlot.tabManager.setOptions({autohideTabBar: true});
+  $(target).off();
+  $(target).on('heatMapLoaded', function (e, heatMap) {
+
+    // Remove verbose tab atop Morpheus dot plot
+    var tabItems = dotPlot.tabManager.getTabItems();
+    window.dotPlot.tabManager.setActiveTab(tabItems[1].id);
+    window.dotPlot.tabManager.remove(tabItems[0].id);
+
+    renderDotPlotLegend();
+
+    // Remove "Options" toolbar button until legend can be updated upon
+    // changing default options for size and color (SCP-1738).
+    // setTimeout is a kludge, but seemingly the only way to do this.
+    setTimeout(function() {
+      var options = $('[data-action="Options"]');
+      options.next('.morpheus-button-divider').remove();
+      options.remove();
+      console.log('removed options')
+    }, 50);
+  });
+
+  // Set render variable to true for tests
+  $(target).data('morpheus', dotPlot);
+  $(target).data('rendered', true);
+  console.log('render status of ' + target + ' at end: ' + $(target).data('rendered'));
+}
+
+function drawDotplot(height) {
+  $(window).off('resizeEnd');
+
+  // Clear out previous stored dotplot object
+  $('#dot-plot').data('dotplot', null);
+
+  // If height isn't specified, pull from stored value, defaults to 500
+  if (height === undefined) {
+    height = $('#dot-plot').data('height');
+  }
+
+  // Pull fit type as well, defaults to ''
+  var fit = $('#dot-plot').data('fit');
+
+  var selectedAnnot = $('#annotation').val();
+  var annotName = selectedAnnot.split('--')[0];
+  var annotType = selectedAnnot.split('--')[1];
+
+  var cluster = $('#cluster').val();
+  $('#search_cluster').val(cluster);
+  $('#search_annotation').val(''); // clear value first
+  $('#search_annotation').val(selectedAnnot);
+
+  var newAnnotPath = dotPlotAnnotPathBase + '?cluster=' + cluster + '&annotation=' + selectedAnnot + '&request_user_token=' + requestToken;
+
+  var renderUrlParams = getRenderUrlParams();
+  // Get annotation values to set color values in Morpheus and draw dotplot in callback
+  $.ajax({
+    url: dotPlotAnnotValuesPath + '?' + renderUrlParams,
+    dataType: 'JSON',
+    success: function(annotations) {
+      renderMorpheusDotPlot(dataPath, newAnnotPath, annotName, annotType, '#dot-plot', annotations, fit, height);
+    }
+  });
+}
+
+$('#annotation').change(function() {
+  $('#dot-plot').data('rendered', false);
+  if ($(this).attr('id') === 'annotation') {
+    var an = $(this).val();
+    // Keep track for search purposes
+    $('#search_annotation').val(an);
+    $('#gene_set_annotation').val(an);
+  }
+  drawDotplot();
+});
+
+// When changing cluster, re-render annotation options and call render function
+$('#cluster').change(function(){
```

Comment:
See previous comment about event listeners.  I understand why you want to abstract this out to different .js files, but I worry about how that's going to impact what we already have (co-mingled JS in HTML views that define event listeners as needed).  Not necessarily saying this has to change, but I'd like a little more insight on this.

Reply:
My [reply](https://github.com/broadinstitute/single_cell_portal_core/pull/278#discussion_r298204507) to your previous comment addresses this case, too.